### PR TITLE
add guidance on merging changes with failing status checks

### PR DESCRIPTION
**Pull request by jdkandersson**
Added guidance on how to merge a PR with failing status checks

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -368,6 +368,22 @@ Alternatives to merging the PR with failing status checks include:
   is lost.
 * Wait for an upstream fix for the issue.
 
+If it is deemed that a change should land despita a failing status check, the
```
Comment (mthaddon) [inline CONTRIBUTING.md:371]: s/despita/despite/

Comment (jdkandersson) [reply CONTRIBUTING.md]: thanks, fixed

**Review by yanksyoon — DISMISSED** [review]
LGTM

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -368,6 +368,22 @@ Alternatives to merging the PR with failing status checks include:
   is lost.
 * Wait for an upstream fix for the issue.
 
+If it is deemed that a change should land despite a failing status check, the
+following artifacts should be added to the PR:
+
+* If the status check can be run in another way, e.g. locally, a screenshot with
```
Comment (nrobinaubertin) [inline CONTRIBUTING.md:374]: I'm not sure to understand what's the goal of the screenshot (it surely not is to guard against bad faith actors)

**Review by arturo-seijas — COMMENTED** [review]
Unfortunately, if test do not pass on PR submission they are unlikely to pass once merge, blocking the release of the changes.

File: CONTRIBUTING.md (hunk 2 above)
Comment (jdkandersson) [reply CONTRIBUTING.md:374]: It is not intended as a guard against bad faith (it coincidentally helps in that case as well). The primary reason is is that on a normal PR, the information on the PR includes the result of the test run. If the test run doesn't work on the repo, there are missing artefacts that would normally be available. This is a way to make them available

**Comment by jdkandersson** [comment]
> Unfortunately, if test do not pass on PR submission they are unlikely to pass once merge, blocking the release of the changes.

Any ideas on how we could resolve this? Some ideas:
* The PR disables the failing status checks
* The charm is updated manually on charmhub

File: CONTRIBUTING.md (hunk 2 above)
Comment (nrobinaubertin) [reply CONTRIBUTING.md:374]: I would suggest copying the text of the result instead of a screenshot in that case

Comment (jdkandersson) [reply CONTRIBUTING.md]: Good suggestion, it is now more generic referring to terminal output

**Comment by nrobinaubertin** [comment]
> Any ideas on how we could resolve this? Some ideas:

Could we have a "force release" command ? It could be a special keyword part of a commit message for example.

**Comment by jdkandersson** [comment]
> > Any ideas on how we could resolve this? Some ideas:
> 
> Could we have a "force release" command ? It could be a special keyword part of a commit message for example.

That would not be easy to implement. The easiest thing is probably to do a manual release after running the charm tests locally in these cases

**Review by yanksyoon — APPROVED** [review]

**Review by merkata — APPROVED** [review]

**Pull request merged**